src/control/server/mgmt_pool_test.go

```diff
@@ -719,37 +719,42 @@ func TestServer_MgmtSvc_PoolCreateDownRanks(t *testing.T) {
 
 func TestServer_MgmtSvc_PoolDestroy(t *testing.T) {
 	log, buf := logging.NewTestLogger(t.Name())
-	missingSB := newTestMgmtSvc(t, log)
-	missingSB.harness.instances[0].(*EngineInstance)._superblock = nil
-	notAP := newTestMgmtSvc(t, log)
-	testPoolService := &system.PoolService{
-		PoolLabel: "test-pool",
-		PoolUUID:  uuid.MustParse(mockUUID),
-		Replicas:  []ranklist.Rank{0, 1, 2},
-		State:     system.PoolServiceStateReady,
-		Storage: &system.PoolServiceStorage{
-			CreationRankStr: ranklist.MustCreateRankSet("0-7").String(),
-		},
+	testPoolService := func() *system.PoolService {
+		return &system.PoolService{
+			PoolLabel: "test-pool",
+			PoolUUID:  uuid.MustParse(mockUUID),
+			Replicas:  []ranklist.Rank{0, 1, 2},
+			State:     system.PoolServiceStateReady,
+			Storage: &system.PoolServiceStorage{
+				CreationRankStr: ranklist.MustCreateRankSet("0-7").String(),
+			},
+		}
 	}
-	svcWithState := func(in *system.PoolService, state system.PoolServiceState) (out *system.PoolService) {
-		out = new(system.PoolService)
+	ready := system.PoolServiceStateReady
+	destroying := system.PoolServiceStateDestroying
```

Comment:
I think we need test cases in which pool state starts out as Creating, which this PR aims to resolve (problem was that a pool destroy --force still was unable to destroy such a pool). Can it be represented in this framework?

1. PoolDestroy request with Recursive: true and Force:false. Expect evict dRPC failure with -DER_AGAIN, and PoolDestroy response with the same error. evict fails since there is no pool service alive (previous create attempt didn't create pool service entirely, and got interrupted with CTRL-C of dmg, and termination of servers). destroy dRPC doesn't happen because force=false.
2. case 1 with Recursive: false and Force:false - same result I think
3. PoolDestroy request with Recursive: true and Force:true. Expect evict dRPC failure, which implementation shall ignore, and pool destroy dRPC success.
4. case 3 with Recursive: true - same result I think

Reply:
sounds good, could you add those cases or do you want me to?

Reply:
I can add them.

Reply:
adding...

Reply:
There is a discrepancy between error handling in ListCont and PoolEvict in the case of creating pool service state. The former returns the error in response whereas the latter returns the error outright.